### ISSUE TITLE
jirashell missing dependency on IPython

### ISSUE THREAD

**Issue by marco-m** [pycontribs/jira#485]
Hello,

although `pip install jira` installs `jirashell`, it doesn't install IPython.

Environment:
```
MacOS 10.13.1 (latest)
virtualenv, python 3.6

virtualenv-36 ~/env/py3
activate ~/env/py3/bin/activate.fish
pip install jira
...

pip show jira
Name: jira
Version: 1.0.10
...
```

What happens:
```
jirashell -s http://jira.atlassian.com
Traceback (most recent call last):
  File "/Users/foo/env/py3/bin/jirashell", line 11, in <module>
    sys.exit(main())
  File "/Users/foo/env/py3/lib/python3.6/site-packages/jira/jirashell.py", line 252, in main
    from IPython.frontend.terminal.embed import InteractiveShellEmbed
ModuleNotFoundError: No module named 'IPython'
```

Either `pip install jira` should install automatically also IPython or it should not install jirashell and the documentation should mention it.

**Comment by hdost** [comment]
Jira Shell is more considered an optional and therefore IPython as many people use this as a library. I do agree though that we could add documentation to mention that if someone wants to use Jira Shell that IPython needs to be installed.

**Comment by ssbarnea** [comment]
If you want to install all dependencies use `pip install jira[cli,opt]`

**Comment by marco-m** [comment]
@ssbarnea 
please re-read my description:

> Either `pip install jira` should install automatically also IPython or it should not install jirashell and > the documentation should mention it.

I know I can do `pip install jira[cli,opt]`, my point is that `pip install jira` installs too much and not enough at the same time.

**Comment by ssbarnea** [comment]
This sounds a little bit as having a cake and eating it kind of issue. Can we find a way that will suit everyone? How would you describe the disired behaviour?

**Comment by marco-m** [comment]
In my understanding, the minimal change that will not make the default install bigger is to remove jirashell from `pip install jira`.

**Comment by ssbarnea** [comment]
Fixed in latest release (1.0.14), where you can assure you have ipython by installing it using `pip install jira[cli]`

**Comment by marco-m** [comment]
merci!
